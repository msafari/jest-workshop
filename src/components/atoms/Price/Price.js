import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
// import { Text } from 'pcln-design-system'

const propTypes = {
  price: PropTypes.number.isRequired
}

const Price = ({
  price
}) => {
  return (
    <PriceAmount>
      {price}
    </PriceAmount>
  )
}

const PriceAmount = styled.span`
  display: inline-block;
  font-size: 12px;
  color: ${props => props.theme.green};
`

// const Price = ({
//   price
// }) => {
//   return (
//     <Text.span
//       color='green'
//       fontSize={0}
//     >
//       {price}
//     </Text.span>
//   )
// }

Price.propTypes = propTypes

export default Price
