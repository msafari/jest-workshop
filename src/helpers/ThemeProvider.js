import React from 'react'
import PropTypes from 'prop-types'
import styled, { ThemeProvider } from 'styled-components'

const propTypes = {
  children: PropTypes.any,
  theme: PropTypes.object
}

const DetailContainer = styled.div`
  width: 100%;
  margin: 0;
  padding: 0;
  font-family: ${props => props.theme.font};
  line-height: 1.4;
  * {
    box-sizing: border-box;
  }
`

const HotelThemeProvider = ({
  children,
  theme
}) => (
  <ThemeProvider theme={theme}>
    <DetailContainer>
      {children}
    </DetailContainer>
  </ThemeProvider>
)

HotelThemeProvider.propTypes = propTypes

export default HotelThemeProvider
