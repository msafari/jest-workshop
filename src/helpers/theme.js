import { theme, legacyTheme } from 'pcln-design-system'

const hotelColors = Object.assign({
}, theme.colors, legacyTheme.colors)

const defaultTheme = Object.assign({
}, theme, {
  colors: hotelColors,
  font: `'Helvetica Neue',Helvetica,Arial,sans-serif`
})

const montserratTheme = Object.assign({
}, theme, {
  colors: hotelColors
})

export { defaultTheme, montserratTheme }
