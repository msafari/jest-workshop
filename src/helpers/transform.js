function transformHotel (hotel) {
  if (hotel.rooms) {
    const formattedRooms = hotel.rooms.map(room => {
      room.defaultCurrency = '$'
      return room
    })
    return Object.assign({}, hotel, {
      rooms: formattedRooms,
      successMessage: 'There are rooms.'
    })
  } else {
    return Object.assign({}, hotel, {
      errorMessage: 'There is no rooms.'
    })
  }
}

export {
  transformHotel
}
