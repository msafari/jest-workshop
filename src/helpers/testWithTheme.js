import React from 'react'
import { shallow, mount } from 'enzyme'
import render from 'react-test-renderer'
import { defaultTheme } from './theme'
import ThemeProvider from './ThemeProvider'

export function renderWithTheme (children) {
  return render.create(
    <ThemeProvider theme={defaultTheme}>
      {children}
    </ThemeProvider>
  )
}

export function shallowWithTheme (children) {
  return shallow(
    <ThemeProvider theme={defaultTheme}>
      {children}
    </ThemeProvider>
  )
}

export function mountWithTheme (children) {
  return mount(
    <ThemeProvider theme={defaultTheme}>
      {children}
    </ThemeProvider>
  )
}
