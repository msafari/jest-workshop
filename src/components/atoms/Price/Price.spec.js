import React from 'react'
import { renderWithTheme } from 'helpers/testWithTheme'
import Price from '.'

test('Price Rendering', () => {
  const price = renderWithTheme(
    <Price price={100} />
  )
  expect(price).toMatchSnapshot()
})
