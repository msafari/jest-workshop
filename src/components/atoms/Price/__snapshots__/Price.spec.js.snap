// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Rendering 1`] = `
.c0 {
  width: 100%;
  margin: 0;
  padding: 0;
  font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  display: inline-block;
  font-size: 12px;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    100
  </span>
</div>
`;
